embedded/store/ongoing_tx.go

```diff
@@ -273,7 +274,12 @@ func (tx *OngoingTx) GetWithFilters(key []byte, filters ...FilterFn) (ValueRef,
 			filters: filters,
 		}
 
+		if tx.readsetSize == tx.st.mvccReadSetLimit {
```

Comment:
you could use a method on the tx here, something like `isMVCCLimitReached` or something similar, as this check is repeated a lot